### PR TITLE
refactor: allow all filterType options (datatable WEB)

### PR DESCRIPTION
**Pull request by willferreiras**

### CODE REVIEW

**Comment by willferreiras** [comment]
Linked to https://github.com/bullcodeapps/sdk/pull/48

File: packages/web/components/DataTable/index.tsx [hunk 1]
```diff
@@ -246,17 +251,20 @@ export default function DataTable({
       label,
       options: {
         filter,
+        display,
```
Comment (giorgiofellipe) [inline packages/web/components/DataTable/index.tsx:254]:
```suggestion
        display = display.toString(), //hack to attend to lib types
```

File: packages/web/components/DataTable/index.tsx [hunk 2]
```diff
@@ -234,6 +236,9 @@ export default function DataTable({
       name,
       label,
       filter = false,
+      display = 'true',
```
Comment (giorgiofellipe) [inline packages/web/components/DataTable/index.tsx:239]:
```suggestion
      display = true,
```

File: packages/web/components/DataTable/index.tsx [hunk 3]
```diff
@@ -246,17 +251,20 @@ export default function DataTable({
...
       options: {
         filter,
+        display,
         filterList: cFilterList,
         customFilterListOptions: {
           render: filterSelectionRender,
         },
         filterOptions: {
           names: filterNames,
+          ...cFilterOptions,
```
Comment (giorgiofellipe) [inline packages/web/components/DataTable/index.tsx:261]: place this spread at the top of the object, we consider all the specializations as a higher priority...

File: packages/web/components/DataTable/index.tsx [hunk 4]
```diff
@@ -246,17 +251,20 @@ export default function DataTable({
...
         },
         sort,
         sortDirection: getSortDirection(order, name),
         searchable,
         customBodyRender: actions ? actionsColumnRender(actions) : customColumnRender,
+        ...cOptions,
```
Comment (giorgiofellipe) [inline packages/web/components/DataTable/index.tsx:267]: place this spread at the top of the object, we consider all the specializations as a higher priority, placing at the top they won't be overridden...

File: packages/web/components/DataTable/index.tsx [hunk 5]
```diff
@@ -67,6 +66,9 @@ export interface DataTableColumn {
   label: string;
   filter?: boolean;
   filterList?: any[];
+  display?: "true" | "false" | "excluded";
```
Comment (giorgiofellipe) [inline packages/web/components/DataTable/index.tsx:69]:
```suggestion
  display?: boolean | "excluded";
```

**Review by giorgiofellipe — CHANGES_REQUESTED** [review]

**Review by giorgiofellipe — APPROVED** [review]

**Pull request merged**